integration-tests/tests/api/audit-logs/audit-log-record.spec.ts

```diff
@@ -0,0 +1,199 @@
+import { endOfDay, startOfDay } from 'date-fns';
+import { waitFor } from 'testkit/flow';
+import { graphql } from 'testkit/gql';
+import { ProjectType } from 'testkit/gql/graphql';
+import { execute } from 'testkit/graphql';
+import { initSeed } from 'testkit/seed';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+
+const s3Client = new S3Client({
+  endpoint: 'http://127.0.0.1:9000',
+  region: 'auto',
+  credentials: {
+    accessKeyId: 'minioadmin',
+    secretAccessKey: 'minioadmin',
+  },
+  forcePathStyle: true,
+});
+
+const auditLogArray = graphql(`
+  query GetAllAuditLogsArray($selector: OrganizationSelectorInput!) {
+    organization(selector: $selector) {
+      organization {
+        id
+        slug
+        auditLogs {
+          edges {
+            node {
+              __typename
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+test.concurrent('Create multiple Audit Log Records for Organization', async ({ expect }) => {
+  const { ownerToken, createOrg } = await initSeed().createOwner();
+  await waitFor(5000);
+  const { organization, createProject } = await createOrg();
+  await waitFor(5000);
+  await createProject(ProjectType.Single);
+  await waitFor(4000);
+
+  const result = await execute({
+    document: auditLogArray,
+    variables: {
+      selector: {
+        organizationSlug: organization.slug,
+      },
+    },
+    authToken: ownerToken,
+  });
+
+  const auditLogs = result.rawBody.data?.organization?.organization.auditLogs.edges;
+
+  expect(auditLogs?.length).toBe(5);
+  expect(auditLogs?.length).not.toBe(0);
+  expect(
+    auditLogs?.find((log: any) => log.node.__typename === 'OrganizationCreatedAuditLog'),
+  ).toBeDefined();
+  expect(
+    auditLogs?.find((log: any) => log.node.__typename === 'ProjectCreatedAuditLog'),
+  ).toBeDefined();
+  expect(
+    auditLogs?.find((log: any) => log.node.__typename === 'TargetCreatedAuditLog'),
+  ).toBeDefined();
+});
+
+const GetAuditLogs = graphql(`
+  query GetAllAuditLogs($selector: OrganizationSelectorInput!) {
+    organization(selector: $selector) {
+      organization {
+        id
+        slug
+        auditLogs {
+          edges {
+            node {
+              id
+              __typename
+              eventTime
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+test.concurrent('Create Audit Log Record for Organization', async ({ expect }) => {
+  await waitFor(5000);
+  const { createOrg, ownerToken } = await initSeed().createOwner();
+  await waitFor(5000);
+  const firstOrg = await createOrg();
+  waitFor(15000);
+  const result = await execute({
+    document: GetAuditLogs,
+    variables: {
+      selector: {
+        organizationSlug: firstOrg.organization.slug,
+      },
+    },
+    authToken: ownerToken,
+  });
+  const auditLogs = result.rawBody.data?.organization?.organization.auditLogs.edges;
+  waitFor(4000);
```

Comment:
TODO @TuvalSimha Here we can simply try to fetch the result until the auditLogs length is the expected length of edges we should have here.
For each attempt, we can increase a small delay. I think up two 10 retries with a delay of 500ms increasing after each attempt seems like a good start. We will then observe how often it will fail during CI and then adjust accordingly.